### DISCUSSION TITLE
preload_image seems not to work anymore

### DISCUSSION THREAD

**Discussion by FelicityF** [jspsych/jsPsych#2297] [Q&A] [answered]
I've been running several experiments online and the images preloaded correctly. However, I just added a new experiment to Pavlovia and it seems the images are not preloading anymore. So for each trial, there is a delay and the pictures are shown a bit later (and not at the same time). 

This is the code in which I create the variables. For block 1, it's selecting all the Foil 1, Foil 2, Foil 3 and Target images for all trials. 

```
var picture_1 = [];
    for (var i=0; i<block1.length; i++) {
         picture_1.push(block1[i].Foil1);
    }

    var picture_1a = [];
    for (var i=0; i< block1.length; i++) {
          picture_1a.push(block1[i].Foil2);
    }

    var picture_1b = [];
    for (var i=0; i< block1.length; i++) {
        picture_1b.push(block1[i].Foil3);
    }

    var picture_1c = [];
    for (var i=0; i< block1.length; i++) {
        picture_1c.push(block1[i].Target);
    }
```

Then at the end I added: 

```
jsPsych.init({
      timeline: timelineAssigned,
      use_webaudio: false,
      preload_audio: [audio_1, audio_2, audio_test, audio_verb, audio_noun],
      preload_image: [picture_1, picture_1a, picture_1b, picture_1c, picture_2, picture_2a, picture_2b, picture_2c, picture_test, 
      picture_test1, picture_test2, picture_test3, .....],
      ....
    });
```

Any idea what the issue could be?

**Comment by becky-gilbert** [comment, marked as answer]
Hi @FelicityF, could you let us know what version of jsPsych you're using? In [jsPsych v6.3.0](https://github.com/jspsych/jsPsych/releases/tag/v6.3.0), we moved all media preloading from `jsPsych.init` into the [`preload` plugin](https://www.jspsych.org/7.0/plugins/preload/). So if you're using jsPsych v6.3+ then it won't work to use `preload_image` as you're doing in the code above - instead you'll need to add a `preload` trial to your experiment timeline.

If that's not the issue then we'll probably need more information to help you figure it out. Maybe you could share a link to your Pavlovia experiment so that we can take a look?

**Reply by FelicityF** [reply]
@becky-gilbert I'm using jspysch-6.1.0, so that shouldn't be the issue if I understand correctly. 

Here is the pilot link to my experiment: 
https://run.pavlovia.org/FelicityFrinsel/pgg_flex/?__pilotToken=3c59dc048e8850243be8079a5c74d079&__oauthToken=3c5ab30c2682430311f983449b023e634536bc364bdaf656a61a5150d52f8144

**Reply by becky-gilbert** [reply]
Thanks very much for sharing the link! I can see what you mean. It looks to me like your audio preloading is working but the image preloading isn't. 

I think the old image preloading parameter name in `jsPsych.init` is `preload_images` (with the _s_ at the end). Can you try changing that and see if that works?

**Reply by FelicityF** [reply]
I just tried it out and that was indeed the issue! Thank you so much.
